### ISSUE TITLE
9811805 Win32 closing Virtual DJ 8

### ISSUE THREAD

**Issue by bluescreen25** [VDJartnet/VDJartnet#16]
If config and preset windows open (background or foreground), then closing VDJ need long time to close, sometimes crash.
If windows are closed before and then you will close VDJ, VDJ closed in milliseconds without problems.

**Comment by aDifferentJT** [comment]
VirtualDJ doesn't notify the plugin before it is closed so I can't find any way to fix this

**Comment by bluescreen25** [comment]
VDJ waits until plugin is closed but why does it take seconds to close plugin with open windows and why only milliseconds if windows are closed.  In older versions it takes also milliseconds on closing plugin with open windows.....so there must be somthing different to versions above from 05/2018. Sometimes i must use taskmanager to close VDJ. So it must be in releasing plugin and/or windows from plugin. Is closing process of windows in different way as if the plugin will be closed and released ?

**Comment by aDifferentJT** [comment]
In the latest versions it isn't crashing but it still takes longer to close, I think it's waiting for the GC.

**Comment by bluescreen25** [comment]
8f94ca9 testet Win32 today. Very long closing process up to 10-15 seconds. It seems to be closed in a second but VDJ not released out of memory. Found in taskmanager background processes. For this time you cannot start VDJ once more. 
In Delphi there was a function GetForegroundWindow (window handle, which the user is currently working on) . Possible to use it , that you can close/hide config-windows while clicking on VDJ.....  like if GetForegroundWindow handle is not config-window then close/hide config-window. Only an idea. In 9811805 was without open config-windows VDJ closing very fast....or config window always on Top in fix Position, so that User must close window before can close VDJ...only ideas..

**Comment by bluescreen25** [comment]
Both versions running on Development notebook. Today testing on my music notebook. Result : Virtual DJ Crash or cannot start. Need we now more requirements (pre installs) to run dll? (Win32bit dll)

**Comment by flumeware** [comment]
Is the visual c++ runtime installed?

**Comment by bluescreen25** [comment]
Yes,32bit runtime. With compiled version from 05/2018 I am working on music notebook ...all ok.  With help from Jonathan I can compile newest versions...today first test on music notebook. Same problems also on 3rd PC, I tested minutes ago....

**Comment by aDifferentJT** [comment]
Aside from .NET and the Visual C++ runtime I cannot think of any other perquisites at the moment, I’ll do some testing in a fresh VM and get back to you.
